app/models/hyrax/operation.rb

```diff
@@ -11,33 +12,57 @@ class Operation < ActiveRecord::Base
     belongs_to :user, class_name: '::User'
 
     # If this is a batch job (has children), check to see if all the children are complete
+    # @see #fail! when none of the children are in a PENDING nor PERFORMING state but at least one is in a FAILURE state
+    # @see #success! none of the children are in a PENDING, PERFORMING, nor FAILURE state
     def rollup_status
       with_lock do
-        stats = children.pluck(:status)
+        # We don't need all of the status of the children, just need to see
+        # if there is at least one PENDING, PERFORMING, or FAILURE.
+        stats = children.select(:status).distinct.pluck(:status)
```

Comment:
Added comments inline.

The gist is: instead of transferring all of the statuses of the children from the database, to Rails, and then instantiating an array and leveraging the include that checks the whole array, opt instead to rely on the database to minimize the return values to that which is needed.